src/libmongoc/src/mongoc/mongoc-topology-private.h

```diff
@@ -188,10 +188,15 @@ typedef struct _mongoc_topology_t {
    bool bypass_query_analysis;
 #endif
 
-   // Corresponds to extraOptions.csflePath
-   char *csfle_override_path;
-   // Corresponds to extraOptions.csfleRequired
-   bool csfle_required;
+   struct {
+      struct {
+         struct {
+            char *cryptSharedLibPath;
+            bool cryptSharedLibRequired;
+         } extraOptions;
+      } autoOptions;
+   } clientSideEncryption;
+
```

Comment:
I got creative here: I didn't like the ambiguous flattening of the namespacing of options to `snake_case` (e.g. `crypt_shared_lib_path`, so I just wrote a nested object that mimics the structure of the corresponding options in the spec document.

If you're totally grossed out by this, feel free to veto. I'm not committed to it.

Reply:
I like this pattern.